drivers/cime/esm.F90

```diff
@@ -1101,6 +1074,281 @@ end subroutine esm_init_pelayout
+    read(cvalue,*) scol_lat
+    call NUOPC_CompAttributeGet(gcomp, name='single_column_lnd_domainfile', value=single_column_lnd_domainfile, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if ( (scol_lon < scol_spval .and. scol_lat > scol_spval) .or. &
+         (scol_lon > scol_spval .and. scol_lat < scol_spval)) then
+       call shr_sys_abort(subname//' ERROR: '//trim(compname)//' both scol_lon and scol_lat must be greater than -999 ')
+    end if
+
+    ! Set the special value for single column - if pts_lat or pts_lon are equal to the special value
+    ! in the component cap - then single column is not activated
+    write(cvalue,*) scol_spval
+    call NUOPC_CompAttributeSet(gcomp, name='scol_spval', value=trim(cvalue), rc=rc)
+    if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+    if (scol_lon > scol_spval .and.  scol_lat > scol_spval) then
+
+       ! NOTE: currently assume that single column capability is restricted to
+       ! ATM, LND, OCN and ICE components only
+       ! verify that WAV and LND are not trying to use single column mode
+       if (trim(compname) == 'WAV' .or. trim(compname) == 'ROF' .or. trim(compname) == 'GLC') then
+          call shr_sys_abort(subname//' ERROR: '//trim(compname)//' does not support single column mode ')
+       end if
+
+       ! ensure that single column mode is only run on 1 pet
+       call ESMF_GridCompGet(gcomp, vm=vm, rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+       call ESMF_VMGet(vm, petcount=petcount, rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       if (petcount > 1) then
+          call shr_sys_abort(subname//' ERROR: single column mode must be run on 1 pe')
+       endif
+
+       write(logunit,'(a,2(f10.5,2x))')trim(subname)//' single column point for '//trim(compname)//&
+            ' has lon and lat = ',scol_lon,scol_lat
+
+       ! This is either a single column or a single point so add attributes
+       call NUOPC_CompAttributeAdd(gcomp, &
+            attrList=(/'scol_area   ', &
+                       'scol_ni     ', &
+                       'scol_nj     ', &
+                       'scol_lndmask', &
+                       'scol_lndfrac', &
+                       'scol_ocnmask', &
+                       'scol_ocnfrac', &
+                       'scol_spval  '/), rc=rc)
+       if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+       if (trim(single_column_lnd_domainfile) /= 'UNSET') then
+
+          ! In this case the domain file is not a single point file - but normally a
+          ! global domain file where a nearest neighbor search will be done to find
+          ! the closest point in the domin file to scol_lon and scol_lat
+
+          status = nf90_open(single_column_lnd_domainfile, NF90_NOWRITE, ncid)
+          if (status /= nf90_noerr) then
+             call shr_sys_abort (trim(subname) //': Cannot open '//trim(single_column_lnd_domainfile))
+          endif
+          status = nf90_inq_dimid (ncid, 'ni', dimid)
```

Comment:
Should check status after each nf90 call

Reply:
Done.